### PR TITLE
[MLU] add coalesce_tensor kernel

### PR DESCRIPTION
**Pull request by Ashelly1**
add coalesce_tensor kernel

### CODE REVIEW

**Comment by Ashelly1** [comment]
image

File: backends/mlu/kernels/coalesce_tensor_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+      if (output[i]->initialized()) {
+        phi::Copy(dev_ctx, *output[i], dev_ctx.GetPlace(), false, &sub_tensor);
+      }
+      offset += use_align
+                    ? custom_kernel::Alignment(
+                          len * size_of_dtype, dev_ctx.GetPlace(), align_size) /
+                          size_of_dtype
+                    : len;
+    }
+  }
```
Comment (ShawnNew) [inline backends/mlu/kernels/coalesce_tensor_kernel.cc:301]: else, set zero

**Review by ShawnNew — CHANGES_REQUESTED** [review]

File: backends/mlu/kernels/coalesce_tensor_kernel.cc (hunk 1 above)
Comment (Ashelly1) [reply backends/mlu/kernels/coalesce_tensor_kernel.cc:301]: 已修改

**Review by ShawnNew — APPROVED** [review]
LGTM

**Pull request merged**